### ISSUE TITLE
Todo list (Dashboard / Full Screen View) - Connect to backend API

### ISSUE THREAD

**Issue by yzho976** [UOA-SE701-Group3-2021/3Lancers#63]
**Is your feature request related to a problem? Please describe.**
The habit tracker is only tested with dummy data and storage. The data need to persist and retrieved from the database.

**Describe the solution you'd like**
Connect the widget with the backend API that will allow data storage and retrieval.

**Describe alternatives you've considered**
Keep dummy data as the default value in case the server failure or loss of network connection.

**Additional context**
N/A

**Comment by khal745** [comment]
Future work: Currently the connection to the backend works for the journal version of the todo list (#219), so likely using a very similar way, but would be blocked by #190. Depending on how the next group wishes to implement the endpoint. Original intention would be return at least a month worth of data, but which events to be displayed (ie the timeframe to collect them) is up to the next group.
